addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,460 @@
+            GregorianCalendar calendar = new GregorianCalendar();
+            calendar.setTime(nextTimer);
+            return new DateTimeType(calendar);
+        } catch (ParseException e) {
+            logger.debug("Could not parse date: {}T{} with pattern 'dd.MM.yy'T'HH:mm:ss'", date, time, e);
+            return UnDefType.UNDEF;
+
+        }
+    }
+
+    private void refreshVersionInfo() throws InterruptedException {
+
+        VersionInfo info = robonectClient.getVersionInfo();
+        if (info.isSuccessful()) {
+            updateState(CHANNEL_VERSION_VERSION, new StringType(info.getRobonect().getVersion()));
+            updateState(CHANNEL_VERSION_COMPILED, new StringType(info.getRobonect().getCompiled()));
+            updateState(CHANNEL_VERSION_COMMENT, new StringType(info.getRobonect().getComment()));
+            updateState(CHANNEL_VERSION_SERIAL, new StringType(info.getRobonect().getSerial()));
+        } else {
+            logger.debug("Could not retrieve mower version info. Robonect error response message: {}",
+                    info.getErrorMessage());
+        }
+
+    }
+    
+    private void refreshLastErrorInfo() throws InterruptedException{
+        ErrorList errorList = robonectClient.errorList();
+        if(errorList.isSuccessful()){
+            if(errorList.getErrors() != null && errorList.getErrors().size() > 0) {
+                ErrorEntry lastErrorEntry = errorList.getErrors().get(0);
+                updateLastErrorChannels(lastErrorEntry);
+            }
+        }else {
+            logger.debug("Could not retrieve mower error list. Robonect error response message: {}",
+                                errorList.getErrorMessage());
+        }
+    }
+
+    private void updateLastErrorChannels(ErrorEntry error) {
+        if (error.getErrorMessage() != null) {
+            updateState(CHANNEL_LAST_ERROR_MESSAGE, new StringType(error.getErrorMessage()));
+        }
+        if (error.getErrorCode() != null) {
+            updateState(CHANNEL_LAST_ERROR_CODE, new DecimalType(error.getErrorCode().intValue()));
+        }
+        if (error.getDate() != null) {
+            State dateTime = convertDateTimeType(error.getDate(), error.getTime());
+            updateState(CHANNEL_LAST_ERROR_DATE, dateTime);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        try {
+            RobonectConfig robonectConfig = getConfigAs(RobonectConfig.class);
+            RobonectEndpoint endpoint = new RobonectEndpoint(robonectConfig.getHost(), robonectConfig.getUser(),
+                    robonectConfig.getPassword());
+            httpClient = new HttpClient();
+            httpClient.start();
+            robonectClient = new RobonectClient(httpClient, endpoint);
+            refreshVersionInfo();
+            Runnable runnable = new Runnable() {
+                
+                private long offlineSince = -1;
```

Comment:
Do you really need/want internal state inside the anonymous inner class runnable?

Reply:
ok... refactored from anonymous to inner class..